flow/src/main/java/com/sequenceiq/flow/core/chain/FlowChainHandler.java

```diff
@@ -60,23 +61,29 @@ public void accept(Event<? extends Payload> event) {
     }
 
     public void restoreFlowChain(String flowChainId) {
-        Optional<FlowChainLog> chainLog = flowLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowChainId);
-        if (chainLog.isPresent()) {
-            String flowChainType = chainLog.get().getFlowChainType();
-            Queue<Selectable> queue = (Queue<Selectable>) JsonReader.jsonToJava(chainLog.get().getChain());
-            Payload triggerEvent = tryDeserializeTriggerEvent(chainLog.get());
+        Optional<FlowChainLog> chainLogOpt = flowLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowChainId);
+        if (chainLogOpt.isPresent()) {
+            FlowChainLog chainLog = chainLogOpt.get();
+            String flowChainType = chainLog.getFlowChainType();
+            Queue<Selectable> queue;
+            if (null != chainLog.getChainJackson()) {
```

Comment:
The entity field `FlowChainLog.chainJackson` is marked as `nullable = false`, so I wonder how this can work. Perhaps checking for `Strings.notEmpty()` would be better?

Reply:
if the `chainJackson` is null that means that this chain was created with a version of CB that did not have this column